### PR TITLE
Compile `wasmer-api` crate to `wasm32-unknown-unknown`

### PR DESCRIPTION
**Pull request by xdoardo**
This patch resolves compiler errors arising when compiling the `wasmer-api` (a.k.a. `backend-api`) crate to `wasm32-unknown-unknown` targets. 

The main complication was the explicit use of a type (`Pin<Box<dyn Stream<Item = Result<Vec<DeployApp>, anyhow::Error>> + Send + Sync>>`) that required a `Sync` that was not available in `wasm32`.

### CODE REVIEW

File: lib/backend-api/src/client.rs [hunk 1]
```diff
@@ -53,11 +53,18 @@ impl WasmerClient {
     }
 
     pub fn new(graphql_endpoint: Url, user_agent: &str) -> Result<Self, anyhow::Error> {
+        #[cfg(target_family = "wasm")]
```
Comment (theduke) [inline lib/backend-api/src/client.rs:56]: Timeout is available when compiling to `wasix`, which might be useful for certain things, so let's be more specific here and limit to the specific `-unknown`  target.

File: lib/backend-api/src/query.rs [hunk 2]
```diff
@@ -1351,10 +1347,17 @@ fn get_app_logs(
...
                         /*
-                            TODO: the resolution of watch should be configurable
-                            TODO: should this be async?
-                        */
+                         * [TODO]: The resolution here should be configurable.
+                         */
+
+                        // No tokio::time::sleep on wasm32 as of now.
+                        // Probably not ideal?
+                        #[cfg(target_family = "wasm")]
```
Comment (theduke) [inline lib/backend-api/src/query.rs:1355]: As above.

File: lib/backend-api/src/client.rs (hunk 1 above)
Comment (xdoardo) [reply lib/backend-api/src/client.rs]: [5dcc9f8](https://github.com/wasmerio/wasmer/pull/4920/commits/5dcc9f8229a53b065ff1a440ccb382fecbc90bb3)

File: lib/backend-api/src/query.rs (hunk 2 above)
Comment (xdoardo) [reply lib/backend-api/src/query.rs]: [5dcc9f8](https://github.com/wasmerio/wasmer/pull/4920/commits/5dcc9f8229a53b065ff1a440ccb382fecbc90bb3)

**Review by theduke — APPROVED** [review]

**Pull request merged**